### PR TITLE
vMIME process

### PR DESCRIPTION
**Pull request by Mathieu-Lala**
* Adding mime processing between vSMTP (connection+filter) and vDelivery (relay or deliver)

TODO:
* postq operation
* flush the queue at the startup

### CODE REVIEW

File: src/resolver/deliver_queue.rs [hunk 1]
```diff
@@ -23,19 +22,28 @@ impl Queue {
             Queue::Dead => "dead",
         }
     }
+
+    pub fn to_path(
+        &self,
+        parent: impl Into<std::path::PathBuf>,
+    ) -> std::io::Result<std::path::PathBuf> {
+        let dir = parent.into().join(self.as_str().to_string() + "/");
```
Comment (Mathieu-Lala) [inline src/resolver/deliver_queue.rs:30]: remove .to_string() + "/"

File: src/main.rs [hunk 2]
```diff
@@ -51,82 +170,32 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
         }
     }
 
-    // the leak is needed to pass from &'a str to &'static str
-    // and initialize the rule engine's rule directory.
```
Comment (ltabis) [inline src/main.rs:55]: Don't remove comments please

File: src/resolver/deliver_queue.rs [hunk 3]
```diff
@@ -23,19 +22,28 @@ impl Queue {
...
+        }
+        Ok(dir)
+    }
 }
 
 /// used to write mail to the delivery queue and send a notification
 /// to the delivery process.
 pub struct DeliverQueueResolver {
-    deliver_proc: crossbeam_channel::Sender<String>,
+    sender: crossbeam_channel::Sender<String>,
```
Comment (ltabis) [inline src/resolver/deliver_queue.rs:41]: Why did you changed the name ?

**Review by ltabis — CHANGES_REQUESTED** [review]

File: src/resolver/deliver_queue.rs (hunk 3 above)
Comment (Mathieu-Lala) [reply src/resolver/deliver_queue.rs:41]: It is not deliver_proc but working_proc. Just sender is more simple

**Review by ltabis — APPROVED** [review]

**Pull request merged**